turbomoleio/core/molecule.py

```diff
@@ -511,6 +605,414 @@ def from_dict(cls, d):
+                raise ValueError('The $periodic data group should be set for periodic systems.')
+            if cell_str is None :
+                raise ValueError('The $cell data group should be set for periodic systems.')
+            mol, fi = get_mol_and_indices_frozen(coordinates_str, cell_str, periodic_str,
+                                                 periodic_extension=periodic_extension)
+            periodicity = int(periodic_str.strip())
+
+            int_def_str = dg.sdg("$intdef", strict=True)
+            if int_def_str is not None:
+                raise ValueError('Internal definitions for periodic systems is not supported.')
+
+            user_def_bonds_str = dg.sdg("$user-defined bonds", strict=True)
+            if user_def_bonds_str is not None:
+                raise ValueError('User-defined bonds for periodic systems is not supported.')
+
+            return cls(mol, frozen_indices=fi, periodicity=periodicity)
+
+        else:
+            return cls(Structure.from_str(string, fmt))
+
+    @classmethod
+    def from_file(cls, filepath, fmt=None, periodic_extension=5.0):
+        """
+        Creates an instance from a file. Could be a coord file or any other
+        format supported by pymatgen Molecule.
+
+        Args:
+            filepath (str): path to the file.
+            fmt (str): the format of the data. could be "coord" for Turbomole
+                coord file or any format supported in pymatgen Molecule.
+                If None the code would try to infer it from the file name.
+
+        Returns:
+            An instance of MoleculeSystem.
+        """
+        fname = os.path.basename(filepath)
+        if fmt == "coord" or (fmt is None and fnmatch(fname, "coord")):
+            with open(filepath) as f:
+                return cls.from_string(f.read(), fmt="coord", periodic_extension=periodic_extension)
+        else:
+            return cls(molecule=Structure.from_file(filepath))
+
+    def to_coord_file(self, filepath="coord"):
+        """
+        Writes the system to a coord file.
+
+        Args:
+            filepath (str): the path to the file
+        """
+        with open(filepath, 'wt') as f:
+            f.write(self.to_coord_string())
+
+    def coord_lines(self):
+        """
+        Gives the lines of the $coord datagroup based on the internal Molecule.
+
+        Returns:
+            list: list of strings with the $coord datagroup.
+        """
+
+        return get_coord_lines(self.structure, self.frozen_indices)
+
+    def to_coord_string(self):
+        """
+        Creates the string of a coord file for this system.
+
+        Returns:
+            str: the string representing the system.
+        """
+        lines = ["$coord"]
+        lines.extend(self.coord_lines())
+
+        if self.periodicity is not None:
+            lines.append("$periodic {}".format(self.periodicity))
+            lattice = self.structure.lattice
+            lmat = lattice.matrix
+            if (
```

Comment:
No indeed. We could switch to the lattice. I don't see any particular reason to use $cell instead of $lattice. The only reason I could see is that it is probably easier to humanly read (but not always). Maybe we could have an attribute whether we want to force the use of $cell (e.g. use_cell_dg=False, by default it will use $lattice) instead of $lattice in the write-related methods (to_coord_file and to_coord_string here). In that case, we only make the check when use_cell_dg is True, otherwise, we just use $lattice. What do you think ?

Reply:
the `use_cell_dg` option could be nice to have, but not strictly necessary. Could also be added at a later time, if needed.  
I am not sure if I understand correctly the reply, but in any case some checks should be done since even with `$lattice` the constrain of being along the x axis in 1D and in the xy plane in 2D still hold.